### ISSUE TITLE
When a player click on End Turn, OverTime effects are not deleted.

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#1010]
In GitLab by @thatlonelybugbear on Sep 27, 2022, 24:39

MidiQOL v10.0.11 and v0.9.81 if you have an OverTime effect, like 
`flags.midi-qol.OverTime | Override | turn=end, label=name of the effect, removeCondition=true` (or whatever removeCondition that evaluates to true)
AND the player clicks on End Turn in the combat tracker, the effect will not get deleted.

If the DM changes the combat turn the OverTime effect gets deleted.

First mention here: <https://discord.com/channels/818799768372641864/818801691864203324/1023877940641026140>

**Comment by tposney** [comment]
In GitLab by @thatlonelybugbear on Sep 27, 2022, 24:41

It could affect more situations than just `removeCondition`, but I haven't tested that yet.

**Comment by tposney** [comment]
Turns out this was a bit tricky. The code that watches for the item roll completion (and then to remove the effect) relies on Hooks - midi-qol.RollComplete, but for non-gm clients they have to forward the roll to the GM client and so they never get the midi-qol.RollComplete so never remove the effect.

I'd coded for actually doing the damage via the executeAsGM, but simply did not think about the effect removal. I've moved all of the per effect processing to the GM client and it now seems to work.
